### DISCUSSION TITLE
API for torch.load()

### DISCUSSION THREAD

**Discussion by iprovalo** [deepjavalibrary/djl#2252] [Development]
I need to be able to read a serialized shared non-layer specific tensor (weights) which is saved like so: `torch.save(model.shared.weight, 'weights.bin')`

In Python it can be read: `torch.load('weights.bin')`

Is there a way to load this tensor in DJL?

Thank you!

**Comment by frankfliu** [comment]
We only support load numpy format, you can convert it to numpy first and save it in either .npy or .npz format.

**Reply by iprovalo** [reply]
Thank you, @frankfliu !  What is the API for that?

**Comment by frankfliu** [comment]
You can use:

```
NDArray.decode(NDManager, byte[])
NDManager.decode(byte[])
NDManager.decode(InputStream)

for .npz:
NDList.decode(NDManager, byte[])
NDList.decode(NDManager, InputStream)
```

See examples: https://github.com/deepjavalibrary/djl/blob/master/integration/src/main/java/ai/djl/integration/tests/ndarray/NDArrayOtherOpTest.java#L821

https://github.com/deepjavalibrary/djl/blob/master/api/src/test/java/ai/djl/ndarray/NDListTest.java#L26

**Reply by iprovalo** [reply]
Thank you again, @frankfliu !
